### PR TITLE
[SPARK-12570] [ML] [Doc] DecisionTreeRegressor: provide variance of prediction: user guide update

### PR DESCRIPTION
**Pull request by yanboliang**
Update user guide doc for `DecisionTreeRegressor` providing variance of prediction.

cc @jkbradley

### CODE REVIEW

File: docs/ml-classification-regression.md [hunk 1]
```diff
@@ -535,7 +535,9 @@ The main differences between this API and the [original MLlib Decision Tree API]
 * use of DataFrame metadata to distinguish continuous and categorical features
 
 
-The Pipelines API for Decision Trees offers a bit more functionality than the original API.  In particular, for classification, users can get the predicted probability of each class (a.k.a. class conditional probabilities).
+The Pipelines API for Decision Trees offers a bit more functionality than the original API.  
+In particular, for classification, users can get the predicted probability of each class (a.k.a. class conditional probabilities);
```
Comment (BenFradet) [inline docs/ml-classification-regression.md]: Line ends with ";".

**Comment by BenFradet** [comment]
Lgtm except for one minor comment.

File: docs/ml-classification-regression.md (hunk 1 above)
Comment (yanboliang) [inline docs/ml-classification-regression.md]: This is not the end, it will concat with the next line in the generated doc.

Comment (BenFradet) [inline docs/ml-classification-regression.md]: My bad, I didnt understand the sentence correctly.

**Comment by jkbradley** [comment]
LGTM pending tests
(There's an indentation bug in the table, but it's fine for now.)

**Comment by jkbradley** [comment]
Merging with master

Thank you!

**Pull request closed without merging**